include/onnxruntime/core/common/status.h

```diff
@@ -74,6 +77,39 @@ inline const char* StatusCodeToString(StatusCode status) noexcept {
       return "GENERAL ERROR";
   }
 }
+#ifdef _WIN32
+inline HRESULT StatusCodeToHRESULT(StatusCode status) noexcept {
+    switch (status)
+    {
+    case StatusCode::OK:
+        return S_OK;
+    case StatusCode::FAIL:
+        return E_FAIL;
+    case StatusCode::INVALID_ARGUMENT:
+        return E_INVALIDARG;
+    case StatusCode::NO_SUCHFILE:
+        return __HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND);
+    case StatusCode::NO_MODEL:
+        return __HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND);
+    case StatusCode::ENGINE_ERROR:
+        return E_FAIL;
+    case StatusCode::RUNTIME_EXCEPTION:
+        return E_FAIL;
+    case StatusCode::INVALID_PROTOBUF:
+        return __HRESULT_FROM_WIN32(ERROR_FILE_CORRUPT);
+    case StatusCode::MODEL_LOADED:
+        return __HRESULT_FROM_WIN32(ERROR_INTERNAL_ERROR);
+    case StatusCode::NOT_IMPLEMENTED:
+        return E_NOTIMPL;
+    case StatusCode::INVALID_GRAPH:
+        return __HRESULT_FROM_WIN32(ERROR_FILE_CORRUPT);
+    case StatusCode::EP_FAIL:
+        return __HRESULT_FROM_WIN32(ERROR_INTERNAL_ERROR);
+    default:
+        return E_FAIL;
+    }
+}
```

Comment:
this exists somewhere else, too right? in winml somewhere. can we share it instead of copying it?

Reply:
nm. I see now that you removed it from the winml spot

---
In reply to: [347040246](https://github.com/microsoft/onnxruntime/pull/2406#discussion_r347040246) [](ancestors = 347040246)